charts/k8ssandra/templates/prometheus/service_monitor.yaml

```diff
@@ -6,6 +6,11 @@ metadata:
   labels:
     release: {{ .Release.Name }}
 {{ include "k8ssandra.labels" . | indent 4 }}
+  {{- if .Values.monitoring }}
```

Comment:
This can be simplified as:

```
  {{- if .Values.monitoring.serviceMonitor }}
  namespace: {{ default .Release.Namespace .Values.monitoring.serviceMonitor.namespace }}
  {{- end }}
```

Reply:
I was getting nil errors prior to adding `serviceMonitor: {}` in the default values file, because trying to test the presence of `.Values.monitoring.serviceMonitor.namespace` required to first make sure `.Values.monitoring.serviceMonitor` existed as it was not mandatory.
I'll apply your suggestion since it's the case now and a dual check is unnecessary.